hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/Normalization.scala

```diff
@@ -47,7 +48,15 @@ class Normalization(elaborator: Elaborator)(using raise: Raise, ctx: Ctx):
       case (c1: Pattern.ClassLike, c2: Pattern.ClassLike) => c1.sym === c2.sym
       case (Pattern.Lit(l1), Pattern.Lit(l2)) => l1 === l2
       case (Pattern.Tuple(n1, b1), Pattern.Tuple(n2, b2)) => n1 === n2 && b1 === b2
-      case (_, _) => false
+      case (Pattern.Record(ls1), Pattern.Record(ls2)) =>
+        ls1.zip(ls2)
+           .forall{case ((fieldName1, p1), (fieldName2, p2)) =>
+            fieldName1 == fieldName2 && p1 =:= p2
+        }
```

Comment:
* `lazyZip` is more efficient
* `=:=` is just a less used alternative to `===`
* `==` shouldn't be used.

```suggestion
        ls1.lazyZip(ls2).forall:
          case ((fieldName1, p1), (fieldName2, p2)) =>
            fieldName1 === fieldName2 && p1 === p2
```